### PR TITLE
fix: trailing stop properly works on both long and short positions

### PR DESCRIPTION
**Pull request by andycheng123**
trailing stop of type both is similar to type buy with short position, type sell with long position

### CODE REVIEW

File: pkg/bbgo/exit_trailing_stop.go [hunk 1]
```diff
@@ -90,21 +90,23 @@ func (s *TrailingStop2) checkStopPrice(price fixedpoint.Value, position *types.P
...
-		if roi.Compare(s.MinProfit) >= 0 {
-			Notify("[trailingStop] activated: ROI %f > minimal profit ratio %f", roi.Float64(), s.MinProfit.Float64())
-			s.activated = true
-		}
-	} else if !s.ActivationRatio.IsZero() {
-		ratio, err := s.getRatio(price, position)
-		if err != nil {
-			return err
-		}
+	if (position.IsLong() && s.Side == types.SideTypeSell) || (position.IsShort() && s.Side == types.SideTypeBuy) {
```
Comment (c9s) [inline pkg/bbgo/exit_trailing_stop.go:93]: 不過為何這邊這樣改?

因為 Long 也有可能用 sideBuy

Comment (andycheng123) [reply pkg/bbgo/exit_trailing_stop.go:93]: Long 用 sidebuy 基本上一 activate 就會生效

File: pkg/bbgo/exit_trailing_stop.go [hunk 2]
```diff
@@ -76,10 +76,16 @@ func (s *TrailingStop2) getRatio(price fixedpoint.Value, position *types.Positio
 	switch s.Side {
 	case types.SideTypeBuy:
 		// for short position, it's:
-		//  (avg_cost - price) / price
-		return position.AverageCost.Sub(price).Div(price), nil
+		//  (avg_cost - price) / avg_cost
+		return position.AverageCost.Sub(price).Div(position.AverageCost), nil
 	case types.SideTypeSell:
 		return price.Sub(position.AverageCost).Div(position.AverageCost), nil
+	case types.SideTypeBoth:
```
Comment (c9s) [inline pkg/bbgo/exit_trailing_stop.go:83]: add the default case too?

File: pkg/bbgo/exit_trailing_stop.go [hunk 3]
```diff
@@ -117,6 +123,12 @@ func (s *TrailingStop2) checkStopPrice(price fixedpoint.Value, position *types.P
 			s.latestHigh = fixedpoint.Min(price, s.latestHigh)
 		case types.SideTypeSell:
 			s.latestHigh = fixedpoint.Max(price, s.latestHigh)
+		case types.SideTypeBoth:
```
Comment (c9s) [inline pkg/bbgo/exit_trailing_stop.go:126]: and default case?

**Review by c9s — APPROVED** [review]

**Review by austin362667 — APPROVED** [review]

**Pull request merged**